core/market/decentralized/tests/test_initial_proposal.rs

```diff
@@ -334,4 +335,71 @@ mod tests {
         assert_eq!(events.len(), 0);
         Ok(())
     }
+
+    /// Requestor gets initial proposal from market and counters it. Proposal should be
+    /// propagated to Provider node and added to events.
+    /// Both Offer and Demand are on the same node.
+    /// TODO: Rewrite this test to use proposals generated by matcher instead
+    ///  of injecting them negotiation engine.
+    #[cfg_attr(not(feature = "market-test-suite"), ignore)]
+    #[actix_rt::test]
+    async fn test_counter_initial_proposal() -> Result<(), anyhow::Error> {
+        let network = MarketsNetwork::new("test_query_initial_proposal")
+            .await
+            .add_market_instance("Node-1")
+            .await?;
+
+        let node1 = network.get_node("Node-1");
+        let market1: Arc<MarketService> = network.get_market("Node-1");
+        let identity1 = network.get_default_id("Node-1");
+        let (offer_id, subscription_id) = node1
+            .inject_proposal(&example_offer(), &example_demand())
+            .await?;
+
+        // We expect that proposal will be available as event.
+        let events = market1
+            .requestor_engine
+            .query_events(&subscription_id, 0.0, Some(5))
+            .await?;
+
+        assert_eq!(events.len(), 1);
+        let init_proposal = match events[0].clone() {
+            RequestorEvent::ProposalEvent { proposal, .. } => proposal,
+            _ => panic!("Invalid event Type. ProposalEvent expected"),
+        };
```

Comment:
to consider: check basic properties (eg. the state) of the  `init_proposal`

Reply:
These properties are tested in `test_query_initial_proposal`.
Or maybe you see something more I can test?
This test was meant to test counter proposal properties, not initial proposal.